INSTALL.rst

```diff
@@ -1,95 +1,128 @@
 Installation
 ============
 
-Step-by-step
-------------
+Prepare the environment
+-----------------------
 
-Install `ffmpeg` and ensure that `ffprobe` is in your PATH:
+1. Install ``ffmpeg`` and ensure that ``ffprobe`` is in your PATH:
 
 .. code-block:: console
 
     $ ffprobe
     ... ffprobe version 3.3.3 ...
 
-Prepare the environment. You will need NodeJS v8:
+2. You will need NodeJS v14. Remember you can use `nvm` to help you manage the NodeJS version.
+
+3. Set up a python virtual environment with python version 3.6. Remember to double check your VIRTUAL_ENV environment variable to ensure later scripts run properly:
+
+.. code-block:: console
+
+    $ echo $VIRTUAL_ENV
+
+If you get an empty output from the previous command, run:
+
+.. code-block:: console
+
+    $ export VIRTUAL_ENV=<path_to_your_env>
+
+4. Activate your virtual environment, create a ``src`` folder, change directories to the ``src`` folder you just created and clone this repo:
 
 .. code-block:: console
 
-    $ ./scripts/setup-npm.sh
-    $ mkvirtualenv cds3
-    (cds3)$ cdvirtualenv ; mkdir src ; cd src
+    (cds3)$ mkdir src ; cd src
     (cds3)$ git clone https://github.com/CERNDocumentServer/cds-videos.git
 
+5. Run a script to set up some npm configurations and install some packages:
+
+.. code-block:: console
+
+    (cds3)$ ./scripts/setup-npm.sh
+
 .. note::
 
-    Before starting the installation you can verify you have installed all the
+    (Deprecated) Before starting the installation you can verify you have installed all the
     system requirements using the `Invenio scripts <https://github.com/inveniosoftware/invenio/tree/master/scripts>`_
 
 
-Install production like setup
+6. Downgrade ``setuptools``, install some important libraries and then install production like setup:
 
 .. code-block:: console
 
-    $ workon cds3
-    (cds3)$ cd cds
+    (cds3)$ pip install "setuptools<58.0"
+    (cds3)$ sudo apt install libsasl2-dev libldap2-dev libssl-dev curl
     (cds3)$ pip install -r requirements.pinned.txt
     (cds3)$ pip install -e .
 
-Or to install the latest released versions of all the dependencies
+Or just install the libraries and latest released versions of all the dependencies:
 
 .. code-block:: console
 
-    $ workon cds3
-    (cds3)$ cd cds
+    (cds3)$ sudo apt-get install libsasl2-dev libldap2-dev libssl-dev curl
     (cds3)$ pip install -e .
 
-Build the assets
+7. Build the assets:
 
 .. code-block:: console
 
     (cds3)$ python -O -m compileall .
     (cds3)$ ./scripts/setup-assets.sh
 
-Make sure that ``elasticsearch`` server is running:
+8. (Deprecated) Make sure that ``elasticsearch`` server is running:
 
 .. code-block:: console
 
     $ elasticsearch
     ... version[2.0.0] ...
 
-Create database & user
+9. Install ``docker`` and ``docker-compose`` then set up some user configurations:
+
+.. code-block:: console
+
+    (cds3)$ sudo apt install docker docker-compose
```

Comment:
This is also for Linux only: you could simply write to install `docker` and `docker-compose`, with a link to the official doc from Docker.